inttest/windows/windows_test.go

```diff
@@ -0,0 +1,241 @@
+// SPDX-FileCopyrightText: 2025 k0s authors
+// SPDX-License-Identifier: Apache-2.0
+
+package windows
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type WindowsSuite struct {
+	suite.Suite
+	kc         *kubernetes.Clientset
+	restConfig *rest.Config
+}
+
+func TestWindowsSuite(t *testing.T) {
+	suite.Run(t, new(WindowsSuite))
+}
+
+func (s *WindowsSuite) SetupSuite() {
+	// kubeconfig := "/Users/jnummelin/.kube/win-config"
+	kubeconfig := os.Getenv("KUBECONFIG")
+	s.Require().NotEmpty(kubeconfig, "KUBECONFIG must be set for this test to work")
+	s.T().Logf("Using kubeconfig: %s", kubeconfig)
+	var err error
+	s.restConfig, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	s.Require().NoError(err, "Failed to build kubeconfig")
+	s.kc, err = kubernetes.NewForConfig(s.restConfig)
+	s.Require().NoError(err, "Failed to create Kubernetes client")
+
+	// Get the server address and try to connect to it via go std lib
+	server := s.restConfig.Host
+	s.T().Logf("Trying to connect to Kube API server: %s", server)
+	client := http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	// Test the connection via stdlib client
+	resp, err := client.Get(server + "/version")
+	if err != nil {
+		s.T().Logf("Failed to connect to Kube API server: %v", err)
+	} else {
+		s.T().Logf("Successfully connected to Kube API server, got status: %s", resp.Status)
+	}
+	defer resp.Body.Close()
+
+	// Test the connection using os.Exec for kubectl
+	cmd := exec.Command("kubectl", "version")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		s.T().Logf("Failed to connect to Kube API server via kubectl: %v", err)
+	} else {
+		s.T().Logf("Successfully connected to Kube API server via kubectl, got output: %s", out.String())
+	}
+}
+
+func (s *WindowsSuite) TearDownSuite() {
+	s.T().Log("Cleaning up test resources")
+	// Delete nginx and iis pods and svcs
+	err := s.kc.CoreV1().Pods("default").Delete(context.Background(), "iis", metav1.DeleteOptions{})
+	s.T().Log("Deleted IIS pods, error:", err)
+
+	err = s.kc.CoreV1().Pods("default").Delete(context.Background(), "nginx-linux", metav1.DeleteOptions{})
+	s.T().Log("Deleted Nginx pods, error:", err)
+
+	err = s.kc.CoreV1().Services("default").Delete(context.Background(), "iis-windows-svc", metav1.DeleteOptions{})
+	s.T().Log("Deleted IIS service, error:", err)
+	err = s.kc.CoreV1().Services("default").Delete(context.Background(), "nginx-linux-svc", metav1.DeleteOptions{})
+	s.T().Log("Deleted Nginx service, error:", err)
```

Comment:
All these calls could be registered as test cleanup functions directly after the corresponding resources have been created. Would keep things closer together.